stumpy/aampi.py

```diff
@@ -202,40 +203,48 @@ def _update_egress(self, t):
         self._p_norm_new[0] = (
             np.linalg.norm(self._T[: self._m] - S[: self._m], ord=self._p) ** self._p
         )
-        self._p_norm_new[:] = np.where(
-            self._p_norm_new < config.STUMPY_P_NORM_THRESHOLD, 0, self._p_norm_new
-        )
+
+        mask = self._p_norm_new < config.STUMPY_P_NORM_THRESHOLD
+        self._p_norm_new[mask] = 0
```

Comment:
is "using mask" better than `np.where`?

Reply:
I find it more straightforward to reason with in my head as `mask` is simply a boolean array. So, it is more appropriate IMHO. `np.where` would be much more complicated I think